File: Leetcode_重刷紀錄/0021.Problems1207. Unique Number of Occurrences/Unique Number of Occurrences.cpp
#include <vector>
#include <unordered_map>
#include <unordered_set>
using namespace std;

class Solution {
public:
    bool uniqueOccurrences(vector<int>& arr) {
        unordered_map<int, int> countMap; // 映射來儲存每個數字的計數
        for (int num : arr) {
            countMap[num]++;
        }

        unordered_set<int> occurrencesSet; // 設定儲存唯一出現次數
        for (auto& it : countMap) {
            // 如果我們無法插入它，則存在重複出現計數
            if (!occurrencesSet.insert(it.second).second) {
                return false;
            }
        }

        return true;
    }
};

File: Leetcode_重刷紀錄/0021.Problems1207. Unique Number of Occurrences/Unique Number of Occurrences.py
class Solution:
    def uniqueOccurrences(self, arr: List[int]) -> bool:
        # 統計列表中每個數字出現的次數
        occurrences = Counter(arr)
        # 檢查每個數字出現的次數是否唯一
        return len(occurrences.values()) == len(set(occurrences.values()))